app/src/main/java/org/mozilla/fenix/compose/list/ExpandableListHeader.kt

```diff
@@ -0,0 +1,166 @@
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.PrimaryText
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Wrapper object for the ExpandableListHeader's state logic
+ *
+ * @param expanded Indicates whether the section of content is expanded.
+ * @param expandActionContentDescription The content description for expanding the section.
+ * @param collapseActionContentDescription The content description for collapsing the section.
+ */
+data class ExpandableListHeaderState(
+    val expanded: Boolean,
+    val expandActionContentDescription: String,
+    val collapseActionContentDescription: String,
+)
+
+/**
+ * Wrapper object for an optional sub action in a ExpandableListHeader
+ *
+ * @param iconDrawable Drawable used for the sub action.
+ * @param contentDescription Optional content description for the sub action.
+ * @param onClick Optional lambda for handling clicks.
+ */
+data class SectionHeaderSubAction(
+    @DrawableRes val iconDrawable: Int,
+    val contentDescription: String?,
+    val onClick: () -> Unit = {}
+)
+
+/**
+ * Expandable header for sections of lists
+ *
+ * @param headerText The title of the header.
+ * @param state Indicates whether the section of content is expanded. If null, the Icon will be hidden.
+ * @param headerSubAction Optional right-aligned element with an optional lambda.
+ * @param onClick Optional lambda for handling header clicks.
+ */
+@Composable
+fun ExpandableListHeader(
+    headerText: String,
+    state: ExpandableListHeaderState? = null,
+    headerSubAction: SectionHeaderSubAction? = null,
+    onClick: () -> Unit = {},
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .clickable(onClick = onClick),
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        Spacer(modifier = Modifier.width(16.dp))
+
+        PrimaryText(
+            text = headerText,
+            fontSize = 14.sp,
+            fontFamily = FontFamily(Font(R.font.metropolis_semibold)),
+            maxLines = 1,
+        )
+
+        state?.let {
+            Spacer(modifier = Modifier.width(8.dp))
+
+            Icon(
+                painter = painterResource(
+                    if (state.expanded) R.drawable.ic_chevron_down else R.drawable.ic_chevron_up
+                ),
+                contentDescription = if (state.expanded) {
+                    state.collapseActionContentDescription
+                } else {
+                    state.expandActionContentDescription
+                },
+                modifier = Modifier.size(20.dp),
+                tint = FirefoxTheme.colors.iconPrimary,
+            )
+        }
+
+        Spacer(modifier = Modifier.weight(1f))
+
+        if (headerSubAction == null) {
```

Comment:
This workaround was introduced to ensure the sub action has a minimal tap area. Until Compose 1.1.0 is introduced where it modifies the intrinsic tap area at runtime, we'll have to bake it into the padding of the sub action instead of the top-level `Row`